test/gtest/outer.hpp

```diff
@@ -0,0 +1,236 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "../driver/tensor_driver.hpp"
+#include "cpu_outer.hpp"
+#include "get_handle.hpp"
+#include "random.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/outer.hpp>
+
+struct OuterTestCase
+{
+    size_t N;
+    size_t M;
+    friend std::ostream& operator<<(std::ostream& os, const OuterTestCase& tc)
+    {
+        return os << " N:" << tc.N << " M:" << tc.M;
+    }
+
+    std::vector<size_t> GetInput() { return std::vector<size_t>({N, M}); }
+};
+
+std::vector<OuterTestCase> OuterTestConfigs()
+{ // n c d h w dim nanPropagation
+    // clang-format off
+    return {
+        {128,128},
```

Comment:
If there are cases which cannot pass IsApplicable of 3 solvers, remove and replace them.

Reply:
I replaced the original cases with other cases which pass "IsApplicable" functions.